gpkit/nomials/map.py

```diff
@@ -135,7 +127,7 @@ def sub(self, substitutions, varkeys, parsedsubs=False):
 
         cp = NomialMap()
         cp.units = self.units
-        cp.expmap, cp.csmap = {}, {}
+        cp.expmap, cp.csmap = {}, self.copy()
```

Comment:
csmap is modified during substitution, but had a roundabout way of starting as a copy of self. Since the exps aren't modified, it doesn't need to be created in as laborious of a fashion as cp itself.